hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/RecordLevelIndexSupport.scala

```diff
@@ -143,22 +104,38 @@ class RecordLevelIndexSupport(spark: SparkSession,
     }
   }
 
+  /**
+   * Return true if metadata table is enabled and record index metadata partition is available.
+   */
+  def isIndexAvailable: Boolean = {
+    metadataConfig.enabled && metaClient.getTableConfig.getMetadataPartitions.contains(HoodieTableMetadataUtil.PARTITION_NAME_RECORD_INDEX)
+  }
+}
+
+object RecordLevelIndexSupport {
   /**
    * If the input query is an EqualTo or IN query on simple record key columns, the function returns a tuple of
    * list of the query and list of record key literals present in the query otherwise returns an empty option.
    *
    * @param queryFilter The query that need to be filtered.
    * @return Tuple of filtered query and list of record key literals that need to be matched
    */
-  private def filterQueryWithRecordKey(queryFilter: Expression): Option[(Expression, List[String])] = {
+  def filterQueryWithRecordKey(queryFilter: Expression, recordKeyOpt: Option[String]): Option[(Expression, List[String])] = {
     queryFilter match {
       case equalToQuery: EqualTo =>
-        val (attribute, literal) = getAttributeLiteralTuple(equalToQuery.left, equalToQuery.right).orNull
-        if (attribute != null && attribute.name != null && attributeMatchesRecordKey(attribute.name)) {
-          Option.apply(equalToQuery, List.apply(literal.value.toString))
+        val attributeLiteralTuple = getAttributeLiteralTuple(equalToQuery.left, equalToQuery.right).orNull
+        if (attributeLiteralTuple != null) {
```

Comment:
In what cases does `getAttributeLiteralTuple` return empty Option? Is it when one of the operands is null?